src/main/grammar/SourcePawn.bnf

```diff
@@ -93,83 +99,295 @@
     NEW_KEYWORD="new"
     PUBLIC_KEYWORD="public"
 
+    TRUE_KEYWORD="true"
+    FALSE_KEYWORD="false"
+    NULL_KEYWORD="null"
+
     // SourcePawn-specific keywords
     ENUM_KEYWORD="enum"
     STRUCT_KEYWORD="struct"
+    METHODMAP_KEYWORD="methodmap"
+    PROPERTY_KEYWORD="property"
+    TYPEDEF_KEYWORD="typedef"
+    TYPESET_KEYWORD="typeset"
 
-    // TODO: remove me (for live preview purposes only)
-    SPACE="regexp:[\ \n\r\t\f]"
+    VIEW_AS_KEYWORD="view_as"
+
+    ANY_TYPE="any"
+    VOID_TYPE="void"
+    INT_TYPE="int"
+    FLOAT_TYPE="float"
+    CHAR_TYPE="char"
+    BOOL_TYPE="bool"
+    FUNCTION_TYPE="function"
 
     line_comment="regexp://[^\r\n]*"
+    doc_comment="regexp:/\*\*(.|\n)*?\*/"
     block_comment="regexp:/\*(.|\n)*?\*/"
     preprocessor_comment="regexp:#(assert|define|else|elseif|endif|endinput|error|file|if|include|line|pragma|section|tryinclude|undef)[^\r\n]*"
 
-    integer_literal="regexp:[-+]?\d[_\d]*"
-    float_literal="regexp:[-+]?\d[_\d]*\.\d[_\d]*(e[-+]?\d+)?"
-    hex_literal="regexp:[-+]?0x[a-fA-F0-9_]+"
-    binary_literal="regexp:[-+]?0b[01_]+"
+    integer_literal="regexp:(\b[-+])?\d[_\d]*"
+    float_literal="regexp:(\b[-+])?\d[_\d]*\.\d[_\d]*(e[-+]?\d+)?"
+    hex_literal="regexp:(\b[-+])?0x[a-fA-F0-9_]+"
+    binary_literal="regexp:(\b[-+])?0b[01_]+"
 
     string_literal="regexp:\"(\\.|[^\"])*\""
-    character_literal="regexp:'(\\.|[^\"])'"
+    character_literal="regexp:'(\\.|[^'])'"
```

Comment:
This was a mistake in the initial commits from years ago (also reflected in the lexer def).